view.php

```diff
@@ -51,6 +51,9 @@
 $PAGE->set_url($url);
 require_login($course, true, $cm);
 
+// Could pass in more params here.. maybe the student id?
```

Comment:
I'm not really clear why this is being added when a user doesn't have 'updateown' capability - if a user is not able to update a checklist, then they shouldn't really be adding comments to it. Bump it down a few lines inside the if statement (or maybe even initialise the javascript within the checklist_class() code, in the same bit of code that initialises the rest of the 'update checklist' javascript - look for the call to M.mod_checklist.init ).

As for the comment - this reads like a TODO comment - please make a decision about whether or not the student id is needed (I'm guessing not - as it should always be $USER->id) and then remove the comment as being redundant.

Reply:
Yes, that code and comment shouldn't be in that spot. Thanks for catching that